### PR TITLE
Pass wrapper name when constructing Zivid::Application

### PR DESCRIPTION
**Pull request by RobKleinIkink**
- Modify Singleton to accept constructor arguments for the object it manages
- Use Zivid::Detail::createApplicationForWrapper to construct the Application for the Python wrapper

Part of ZIVID-8484

### CODE REVIEW

**Review by johningve — DISMISSED** [review]
LGTM

**Review by johningve — APPROVED** [review]

File: src/include/ZividPython/Releasable.h [hunk 1]
```diff
@@ -129,12 +129,13 @@ namespace ZividPython
     class Singleton
     {
     public:
-        Singleton()
+        template<typename... Args>
+        Singleton(Args &&...args)
```
Comment (vawale) [inline src/include/ZividPython/Releasable.h:133]: we need to delete the copy/move ctors or restrict the type of `Args` to not be `Singleton`, else this constructor will take precedence over copy/move.

Comment (vawale) [reply src/include/ZividPython/Releasable.h]: pushed a fixup for this

**Review by vawale — DISMISSED** [review]

File: src/include/ZividPython/Releasable.h [hunk 2]
```diff
@@ -129,12 +132,14 @@ namespace ZividPython
     class Singleton
     {
     public:
-        Singleton()
+        template<typename... Args,
+                 typename std::enable_if<!has_type<Singleton<T>, Args...>::value, void *>::type = nullptr>
```
Comment (vawale) [inline src/include/ZividPython/Releasable.h:136]: Possibly also need to remove reference and cv qualifiers from Args

Comment (vawale) [reply src/include/ZividPython/Releasable.h]: fixed

**Review by vawale — APPROVED** [review]

**Pull request merged**